### PR TITLE
C++ ingest string views instead of const char* (almost) everywhere

### LINKED ISSUE #3834: [c++] Consider taking `std::string_view` or similar for strings
Right now the C++ api uses nullterminated `const char*` everywhere which is nice for C (and ultimiately Rust) interop.
However, passing in std::string can get a bit clumsy:
```cpp
rec.log(
        (std::string("world/") + name + "_pt").c_str(),
        rerun::Points3D({pos}).with_colors(color)
    );
```

The modern way is to use `std::string_view` but the problem is that they are not guaranteed to be nullterminated, complicating the FFI - either C needs to get more complex string types as well or we do a temp allocation for this!

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* Fixes #3834 

Makes api nicer to use. Trickles down to C api making us technically more flexible!
The only big drawback is that `std::string_view((const char*)nullptr)` is undefined behavior 🤦 

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/4023) (if applicable)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/4023)
- [Docs preview](https://rerun.io/preview/71fa5ff2bde8d0b8e8ca5da0d1dc4c9182f82c9d/docs) 
- [Examples preview](https://rerun.io/preview/71fa5ff2bde8d0b8e8ca5da0d1dc4c9182f82c9d/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

File: crates/rerun_c/src/rerun.h [hunk 1]
```diff
@@ -18,6 +18,33 @@ extern "C" {
...
 
+/// A Utf8 string with a length in bytes.
+typedef struct rr_string {
+    /// Pointer to a UTF8 string.
+    ///
+    /// Does *not* need to be null-terminated.
+    /// Rerun is guaranteed to not read beyond utf8[length_in_bytes-1].
+    const char* utf8;
+
+    /// The length of the string in bytes, *excluding* null-terminator.
```
Comment (emilk) [inline crates/rerun_c/src/rerun.h:29]: The current wording sort of makes it sounds like there should be a null-terminator after all

```suggestion
    /// The length of the string in bytes (*excluding* null-terminator, if any).
```

**Review by emilk — APPROVED** [review]
LGTM, except for the obvious bug :)

File: crates/rerun_c/src/lib.rs [hunk 2]
```diff
@@ -23,6 +23,21 @@ use re_sdk::{
 // ----------------------------------------------------------------------------
 // Types:
 
+/// `rr_string` (the name `CString` was already taken)
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct CStringWithLength {
```
[the remark this answers is not in the document]
Comment (Wumpf) [reply crates/rerun_c/src/lib.rs:28]: Hmm both are good 🤔

File: rerun_cpp/src/rerun/data_cell.cpp [hunk 3]
```diff
@@ -6,7 +6,7 @@
 namespace rerun {
 
     Result<DataCell> DataCell::create(
-        const char* name, const std::shared_ptr<arrow::DataType>& datatype,
+        std::string name, const std::shared_ptr<arrow::DataType>& datatype,
```
[the remark this answers is not in the document]
Comment (Wumpf) [reply rerun_cpp/src/rerun/data_cell.cpp:9]: nice catch

**Pull request merged**